### PR TITLE
CI: Fix Flake8 linter errors in man/ directory

### PR DESCRIPTION
**Pull request by arohanajit**
### Description:

This PR addresses several linter errors related to wildcard imports and improves overall code quality in multiple Python scripts within the project. The main focus was on resolving F403 and F405 errors, which are related to the use of wildcard imports and potential undefined names.

### Changes made:

   - Replaced wildcard import with specific imports from `build_html`.
   - Removed bare except clauses with Exception keyword

### Rationale for changes:

1. **Resolving linter errors**: The primary motivation was to address F403 and F405 errors. These errors indicate potential issues with undefined names due to wildcard imports.

2. Where applicable, string formatting was updated to use f-strings, which are more readable and efficient in Python 3.

These changes do not alter the functionality of the scripts. They purely focus on code quality improvements and adherence to Python best practices.

### CODE REVIEW

**Comment by echoix** [comment]
I agree with all these changes. However, we need to have a better title, and that brings us to how to better summarize the PR: it touches `man` AND `utils`, where the changes in man changes the star import, and the changes in utils change some exception handling.

It seems to me that it might better be off with two PRs. (Pick one of the two kind of issues and remove it from the PR, and make a new one with the other half)

For the one touching the `man` directory, I see in the .flake8 exclusion that other files have the same exclusions that are fixed here. Are the remaining ones very different from here? If not, please apply the same pattern to these other files (they are very similar to each other, so it should be easy).

**Comment by arohanajit** [comment]
Thanks for pointing that out @echoix. I have removed the utils updates and will put that in a separate PR

**Comment by echoix** [comment]
There are conflicts to solve here too

**Comment by arohanajit** [comment]
Fixed the conflicts

File: .flake8 [hunk 1]
```diff
@@ -20,17 +20,7 @@ per-file-ignores =
...
-    man/parser_standard_options.py: F403, F405
-    man/build_class.py: F403, F405
-    man/build_class_rest.py: F403, F405
-    man/build_check.py: F403, F405
-    man/build_full_index.py: F403, F405
-    man/build_index.py: F403, F405
-    man/build_index_rest.py: F403, F405
-    man/build_keywords.py: F403, F405, E722
-    man/build_topics.py: F403, F405, E722
+    lib/init/grass.py: E722, F821, F841
```
Comment (echoix) [inline .flake8:23]: I don't think you intended to add this back when solving the conflicts

Comment (arohanajit) [reply .flake8]: Yeah, I must've missed it while updating the conflict. I've updated it now

**Review by echoix — APPROVED** [review]

**Pull request merged**